sim-lib/src/cln.rs

```diff
@@ -105,8 +105,16 @@ impl LightningNode for ClnNode {
             })
             .await
             .map_err(|s| {
-                // TODO: Parse CLN rpc status to determine if it's a terminal error or not.
-                LightningError::TerminalError(format!("{:?}", s))
+                let message = s.message();
+                // REF: https://docs.corelightning.org/reference/lightning-keysend#return-value
+
+                // Error codes -1 and 203 indicate terminal errors
+                if message.contains("Some(-1") | message.contains("Some(203") {
+                    return LightningError::TerminalError(format!("{:?}", message));
+                }
+
+                // Error codes, 205, 206 and 210 indicate temporary errors that can be retried
+                LightningError::SendPaymentError(format!("{:?}", message))
```

Comment:
There's no need to return here, given you're evaluating the result with `?` later.

```suggestion
// Error codes -1 and 203 indicate terminal errors
if message.contains("Some(-1") | message.contains("Some(203") {
    LightningError::TerminalError(format!("{:?}", message))
} else {
    // Error codes, 205, 206, and 210 indicate temporary errors that can be retried
    LightningError::SendPaymentError(format!("{:?}", message))
}
```